### ISSUE TITLE
Idea collection for a new demo App

### ISSUE THREAD

**Issue by timunie** [MahApps/MahApps.Metro#3818]
**Is your feature request related to a problem? Please describe.**
Now that the v2.0.0 is out maybe the Demo App should be updated as well. I would like to open this feature request to collect some ideas / drafts

**Describe the solution you'd like**
These are my ideas so far: 
- Use the `HamburgerMenu` instead of a `TabControl`. 
- Use the same Layout for almost every Control
- Provide a Link to the online doc where possible (see: #3053 )
- Provide sample `XAML` to copy & paste it into your own app 

**Additional context**
I have implemented this into my personal Controls-Libary and here is an early preview of it: 
![DemoAppExample](https://user-images.githubusercontent.com/47110241/82902539-e15e8900-9f5f-11ea-9382-f1bb38862eff.png)

To create this view:
```xaml
<local:ExampleView_Base x:Name="demoView" xml:space="preserve">
    <local:ExampleView_Base.ExampleXaml><![CDATA[<ctrls:CircularProgressBar 
        Value="[Value]" 
        Minimum="[Minimum]"
        Maximum="[Maximum]"
        Width="[Width]"
        Height="[Height]"
        HorizontalAlignment="[HorizontalAlignment]"
        VerticalAlignment="[VerticalAlignment]"
        Content="[Content]"
        ContentStringFormat="[ContentStringFormat]" />
        ]]></local:ExampleView_Base.ExampleXaml>
    <ctrls:CircularProgressBar x:Name="MyProgressBar"
                                Width="200"
                                Height="200"
                                HorizontalAlignment="Center"
                                VerticalAlignment="Center"
                                ContentStringFormat="0' %'"
                                Maximum="100"
                                Minimum="0"
                                Value="5" />
</local:ExampleView_Base>
```

and in the Constructor:
```c#
public RadialProgressExample()
{
    InitializeComponent();

    demoView.AddDemoProperty(CircularProgressBar.ValueProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.MinimumProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.MaximumProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.IsFilledProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.IsIndeterminateProperty, MyProgressBar);

    demoView.AddDemoProperty(CircularProgressBar.ContentProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.ContentStringFormatProperty, MyProgressBar);

    demoView.AddDemoProperty(CircularProgressBar.WidthProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.HeightProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.HorizontalAlignmentProperty, MyProgressBar);
    demoView.AddDemoProperty(CircularProgressBar.VerticalAlignmentProperty, MyProgressBar);
}
```

**Closed Issues**
None

Happy coding
Tim

**Comment by timunie** [comment]
Edit: Here is the preview taken from: https://github.com/timunie/TimsWpfControls

**Comment by AKruimink** [comment]
Might be worth having a look at something like ShowMeTheXAML which i think MaterialDesignInXAML uses to do XAML display in their demo (not saying that should be used straith up, but might atleast give some good idea's)

**Comment by timunie** [comment]
@AKruimink good point but I don't know how to modify it to our needs right now. It is on the list 😊

**Comment by amkuchta** [comment]
I love the idea of this. Perhaps instead of having a separate hamburger menu item for each control, though, maybe we could have groups of controls ("buttons", "progress", etc.), with a list view of individual controls on the view for that group? If we have an individual hamburger menu item for each control, we are going to have a lot of hamburger menu items

**Comment by punker76** [comment]
Picking this up again, six years late, and with a direction.

Our demo shows everything at once. `ButtonsExample.xaml` is 566 lines in five columns, and that is one of the tidier pages. It is fine to browse and useless to look something up in: you find a control only if you already know which tab it lives in, and there is no way from a control to its documentation.

What I would like to build instead is along the lines of the WinUI 3 Gallery. Not its looks, it should still look like MahApps, but the way it is laid out:

- one page per control instead of one wall per category
- a navigation with search down the side, which is a job for our own `AutoSuggestBox`
- on a page, several small example cards instead of one big surface
- per card: the living control, a set of options to turn its properties with, and the code below it, collapsed, with a copy button
- per page: a heading, one sentence of description, a link to the docs on mahapps.com and one to the source file (that was #3053)
- a page of its own for theme, accent and the style variants, since that is the part of MahApps worth showing off

The card is the load bearing part. Almost every question people arrive with has the shape "how do I get this control to look like that", and a card answers exactly that, because you can turn the property and take the code with you afterwards.

@timunie, #3825 is closed, and not because it was wrong. What comes across from it: the demo view with the control on one side and its properties on the other, an editor per property type, the grouping of those properties, and the button to the online docs in the header.

One thing gets better on the way. Your `DemoView` kept the shown code as a second copy with `[Value]` placeholders in it, which is a copy somebody has to maintain. ShowMeTheXAML has no second copy, the build writes the markup of every sample into the assembly, but it shows the markup as it stands in the file, without the values the user just set. Its formatter is an interface, so we put our own behind it: take what the build delivered and set the attributes of the watched properties to the values that are currently on. Then the code is always right and nobody maintains it. #4648 puts the package in as the groundwork.

The plan is a project of its own, `MahApps.Metro.Gallery`, next to the current demo rather than instead of it. The page layout cannot be peeled out of 23 tabs one at a time, and the old demo stays usable while the new one grows. It goes away when the gallery covers the controls.

And about the part that killed the last attempt, the forty pages of legwork: those pages will not be written from scratch. The existing `ExampleViews` get taken apart and their pieces moved into cards, which is half an hour per page and can grow without anything being broken in between.

Ideas and objections welcome, especially on where the cards should sit and how the categories should be cut.

**Comment by timunie** [comment]
FluentAvalonia has a really nice Demo. Worth a look if / what we want to copy from them.

**Comment by punker76** [comment]
Had a look, thanks for the pointer. FAControlsGallery's `ControlExample` is the same shape as what I had in mind, which is reassuring, and two things from it are worth taking: the options pane that folds away when the window gets narrow, and working the links out from the target type instead of writing them down per page.

It does share one thing with the WinUI gallery though. The code beside a sample is a `XamlSource` string, with `Substitutions` putting the current values into it, so the shown code is a second copy that somebody keeps in step. That is the copy that goes stale, and it is the one place where we can do better than both of them: ShowMeTheXAML takes the markup out of the build, and a formatter of ours sets the watched properties into it. The code under a card is then the code that actually runs, whatever the reader has just turned.

#4649 has the first of it: navigation with search, the card, three control pages and a settings page for base colour and accent. It is a draft and the shape is the part worth arguing about, so say if the card is cut the way you would cut it.
